### PR TITLE
Add phase alignment constants for each signal

### PR DESCRIPTION
**Pull request by adel-mamin**
The issue https://swift-nav.atlassian.net/browse/EX-361
I could not find the exact alignment corrections listed in RTCM standard. So I used the carrier phase alignment constants according to RINEX v3.03 TABLE A23.

### CODE REVIEW

File: src/signal.c [hunk 1]
```diff
@@ -65,6 +70,7 @@ static const code_table_element_t code_table[CODE_COUNT] = {
                       GPS_L1CA_PRN_PERIOD_MS,
                       GPS_L1CA_CARR_TO_CODE,
                       GPS_L1_DOPPLER_MAX_HZ,
+                      0.f, /* assuming CODE_GPS_L1CA */
```
Comment (mbavaro) [inline src/signal.c]: correct

File: src/signal.c [hunk 2]
```diff
@@ -223,6 +240,7 @@ static const code_table_element_t code_table[CODE_COUNT] = {
                         SBAS_L1CA_PRN_PERIOD_MS,
                         SBAS_L1CA_CARR_TO_CODE,
                         SBAS_L1_DOPPLER_MAX_HZ,
+                        0.f, /* not used */
```
Comment (mbavaro) [inline src/signal.c]: reference signal

File: src/signal.c [hunk 3]
```diff
@@ -249,6 +268,7 @@ static const code_table_element_t code_table[CODE_COUNT] = {
                        SBAS_L5_PRN_PERIOD_MS,
                        SBAS_L5_CARR_TO_CODE,
                        SBAS_L5_DOPPLER_MAX_HZ,
+                       0.f, /* not used */
```
Comment (mbavaro) [inline src/signal.c]: reference signal

File: src/signal.c [hunk 4]
```diff
@@ -262,6 +282,7 @@ static const code_table_element_t code_table[CODE_COUNT] = {
...
```
Comment (mbavaro) [inline src/signal.c]: `-0.25f`

File: src/signal.c [hunk 5]
```diff
@@ -275,6 +296,7 @@ static const code_table_element_t code_table[CODE_COUNT] = {
...
```
Comment (mbavaro) [inline src/signal.c]: aligned with `CODE_SBAS_L5I`

File: src/signal.c [hunk 6]
```diff
@@ -474,6 +510,7 @@ static const code_table_element_t code_table[CODE_COUNT] = {
                       0,
                       0,
                       GAL_E8_DOPPLER_MAX_HZ,
+                      0.f, /* reference signal (see L8I in [1]) */
```
Comment (mbavaro) [inline src/signal.c]: I still don't know how a receiver would generate this `E8I` in practice.. but fair enough..

File: src/signal.c [hunk 7]
```diff
@@ -606,6 +652,7 @@ static const code_table_element_t code_table[CODE_COUNT] = {
                         0,
                         0,
                         BDS3_B1C_DOPPLER_MAX_HZ,
+                        0.f, /* not used (interoperable with SBAS) */
```
Comment (mbavaro) [inline src/signal.c]: here we are in unknown territory, LGTM for now

**Comment by adel-mamin** [comment]
@mbavaro This PR seems to be at odds with the code here: https://github.com/swift-nav/piksi_firmware_private/blob/master/src/acq/manage.c#L1042-L1046. What do you think?

**Comment by mbavaro** [comment]
Not until you actually use this helper function in PFWP :)
Note that we have to be careful on how and where we use this table. 
SBP and RTCM claim to carry already aligned phases.
So does at least also Septentrio SBF.

**Pull request merged**